### ISSUE TITLE
John the ripper is not cracking 

### ISSUE THREAD

**Issue by kimreq** [openwall/john#4249]
@magnumripper 

heyaaa ,while I'm cracking the password i get this message and I'm not even sucessul in cracking it help me 
"warning only 2 candidates left minimum 8 needed for performance"
or someone please explain why I'm getting this error ???
cheers

**Comment by solardiz** [comment]
This is not an error, but is a benign warning.

Our use of GitHub issues is for tracking bugs or planned changes, and your issue isn't that. For user support, please use the john-users mailing list: https://www.openwall.com/lists/john-users/ (and you'd need to describe your use case and actual issue in some detail).

magnum, should we simply close this issue or maybe rename and reuse it for considering ways to mute/postpone printing that warning? The difficulty with it is that at the time we detect the condition we often don't yet know whether the performance impact will be significant or negligible.

**Comment by kimreq** [comment]
@solardiz 
thanks but I'm not able to crack my ssh id_rsa and I'm getting some type of warning messages 
![2020-04-08 11_27_50-Kali - VMware Workstation](https://user-images.githubusercontent.com/36020960/78749613-3896a300-798c-11ea-847f-d5cc9a6ef89e.png)

and at last it's showing 
this
![2020-04-08 11_33_36-Kali - VMware Workstation](https://user-images.githubusercontent.com/36020960/78749930-dd18e500-798c-11ea-8b54-8d095e9c41df.png)

and yet It's not abel to crack password from ssh hash

**Comment by magnumripper** [comment]
Lord give me strength.

Since the OP clearly doesn't read answers (nor documentation) there would be no point in trying to help him so I blocked him.

**Comment by magnumripper** [comment]
> magnum, should we simply close this issue or maybe rename and reuse it for considering ways to mute/postpone printing that warning? The difficulty with it is that at the time we detect the condition we often don't yet know whether the performance impact will be significant or negligible.

Postponing it might be a thing. It would be nice to mute it until we at least got any counts in the status struct. You wanted it even for a single salt and/or a tiny wordlist and that makes it trickier.

**Comment by solardiz** [comment]
@coopjab Like I said, please join the john-users mailing list and post in there for user support. You need to scroll down to bottom of that web page I referenced for how to subscribe. When you post, please use copy-paste from your terminal instead of screenshots.

magnum, I think blocking is too harsh.

Regarding that warning, upon thinking of it some more I don't like postponing it because with a slow (non-)hash on GPU or such we might have just one `crypt_all` invocation for the entire run yet it might be taking a long while and the warning would be helpful to have before that one call rather than after it.

What we may do is mute the warning when the format is fast (add a `FMT_FAST` flag?) and the number of different salts times the number of wasted candidate password slots (`min_keys_per_crypt` minus the actual number of candidates left) is below a threshold (say, 1 million). Unfortunately, this won't help for slow(er) formats like SSH keys.

**Comment by magnumripper** [comment]
I admit having pondered a FMT_FAST flag (or the opposite) before, although it's a bit weird (moves over time).

A wild idea (perhaps too complex though) is we could actually post-pone the message with some logic that prints to a buffer that `signals.c` can detect a couple of seconds later.

Another thing we could consider is to take some rough speed sample during self-tests. That should be doable and could give us plenty of fine "data" to use later. The little work I did with the (almost existing already) "portable hi-res timer" could give fruit.

**Comment by magnumripper** [comment]
I think I got in love with that last idea. Should be fairly easy, and self-adjusting over archs and time.

**Comment by solardiz** [comment]
magnum, this is your repo so ultimately you decide (and I got to hurry up to move this repo to an Openwall account), but can you please unblock @coopjab? I think we shouldn't alienate new users, not even if they don't play 100% by our rules. I think blocking is for deliberate spamming and trolling, not for having difficulty to follow our suggested ways of communication. If someone doesn't read what we say, we (especially you, if that's your preference) can similarly not care to read what they say - but blocking is excessive. I hope you got the strength you prayed for and can use it to ignore rather than block discussions you don't want to participate in. ;-)

> A wild idea (perhaps too complex though) is we could actually post-pone the message with some logic that prints to a buffer that `signals.c` can detect a couple of seconds later.

I can see this result in new complications and issues such as with atomicity and memory ordering and with the message getting printed at a point where it's been already rendered irrelevant by other (non-postponed) messages printed before it (especially in batch mode).

> Another thing we could consider is to take some rough speed sample during self-tests.

I doubt the self-tests are good as benchmarks, even for this purpose where we don't need much precision but rather just the order of magnitude, but if you're in love feel free to give this a try. ;-)

**Comment by magnumripper** [comment]
> can you please unblock @coopjab

M'kay but this, like you said, puts quite some pressure on you to move the repo ASAP. I have done exactly this for years: Clueless windows lusers of age 13 that don't respond at all to kind requests gets banned for life with no further warning or notice.

Please note how I did not treat some other recent young Jedi's bad even though they too were totally in the wrong forum, because they actually responded to instructions or questions. I can go through hoops to help them if they react like responsible citizens.

**Comment by kholia** [comment]
Besides lack of time, interacting with apparently lazy or misbehaving users who don't understand kind requests was demotivational for me. I rather focus my limited free time and energy somewhere else.

I am just reporting what I felt as it is. Not debating or arguing about anything.

Interacting with responsive, responsible users is always a pleasure and I love going the extra miles for them (hopefully evident by so many crazy formats introduced over the years).
